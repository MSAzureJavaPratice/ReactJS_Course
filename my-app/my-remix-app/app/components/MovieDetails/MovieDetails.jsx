import React from 'react';
import PropTypes from 'prop-types';
import './MovieDetails.css';

const MovieDetails = ({ movie }) => {
  if (!movie) {
    return <p className="movie-details-loading">Loading movie details...</p>;
  }

  const { poster_path: posterPath, title, release_date: releaseDate, vote_average: rating, runtime, overview } = movie;

  return (
    <div className="movie-details">
      <img src={posterPath} alt={title} className="movie-details__image" />
      <div className="movie-details__info">
        <h2>{title}</h2>
        <p>Release Date: {releaseDate}</p>
        <p>Rating: {rating}</p>
        <p>Runtime: {runtime} minutes</p>
        <p>{overview}</p>
      </div>
      <a href="/"><button className="back-button">Go Back to Home</button></a>
    </div>
  );
};

// Defining PropTypes
MovieDetails.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number.isRequired,
    poster_path: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    release_date: PropTypes.string.isRequired,
    vote_average: PropTypes.number,
    runtime: PropTypes.number,
    overview: PropTypes.string.isRequired,
  })
};

export default MovieDetails;