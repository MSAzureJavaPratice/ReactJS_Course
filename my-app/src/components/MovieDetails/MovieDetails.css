.movie-details {
    display: flex;
    gap: 20px;
    margin-top: 20px;
}

.movie-details__image {
    width: 300px;
}

.movie-details__info {
    max-width: 600px;
}