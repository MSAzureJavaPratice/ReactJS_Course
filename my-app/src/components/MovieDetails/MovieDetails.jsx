import React, { useEffect, useState } from "react";
import movieService from "../../service/movieService";
import "./MovieDetails.css";

const MovieDetails = ({ movieId }) => {
  const [movie, setMovie] = useState(null);

  useEffect(() => {
    const fetchMovieDetails = async () => {
      try {
        console.log('Fetch Movie Details: ' + movieId);
        const movieData = await movieService.getMovieById(movieId);
        setMovie(movieData);
      } catch (error) {
        console.error("Error fetching movie details:", error);
      }
    };

    if (movieId) {
      fetchMovieDetails();
    }
  }, [movieId]);

  if (!movie) {
    return <div>Loading...</div>;
  }

  const {
    poster_path: posterPath,
    title,
    release_date: releaseDate,
    vote_average: rating,
    runtime,
    overview,
  } = movie;

  return (
    <div className="movie-details">
      <img src={posterPath} alt={title} className="movie-details__image" />
      <div className="movie-details__info">
        <h2>{title}</h2>
        <p>Release Date: {releaseDate}</p>
        <p>Rating: {rating}</p>
        <p>Runtime: {runtime} minutes</p>
        <p>{overview}</p>
      </div>
    </div>
  );
};

export default MovieDetails;
