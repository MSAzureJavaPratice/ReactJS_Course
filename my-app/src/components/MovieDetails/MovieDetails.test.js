import React from "react";
import { render, screen } from "@testing-library/react";
import MovieDetails from "./MovieDetails";

describe("MovieDetails Component", () => {
  const mockMovie = {
    imageUrl: "https://via.placeholder.com/150",
    name: "Inception",
    releaseYear: "2010",
    rating: "8.8/10",
    duration: "148 minutes",
    description: "A thief who steals corporate secrets...",
  };

  test("renders movie details correctly", () => {
    render(<MovieDetails movie={mockMovie} />);

    // Check for rendered content
    expect(screen.getByText("Inception")).toBeInTheDocument();
    expect(screen.getByText("2010")).toBeInTheDocument();
    expect(screen.getByText("Rating: 8.8/10")).toBeInTheDocument();
    expect(screen.getByText("Duration: 148 minutes")).toBeInTheDocument();
    expect(
      screen.getByText("A thief who steals corporate secrets...")
    ).toBeInTheDocument();
  });

  test("renders movie image with correct alt text", () => {
    render(<MovieDetails movie={mockMovie} />);
    const img = screen.getByAltText("Inception");
    expect(img).toHaveAttribute("src", mockMovie.imageUrl);
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<MovieDetails movie={mockMovie} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
