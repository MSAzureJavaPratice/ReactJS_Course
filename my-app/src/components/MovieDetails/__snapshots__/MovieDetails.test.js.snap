// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="movie-details"
  >
    <img
      alt="Inception"
      class="movie-details__image"
      src="https://via.placeholder.com/150"
    />
    <div
      class="movie-details__info"
    >
      <h2>
         Inception 
      </h2>
       
      <p>
         2010 
      </p>
       
      <p>
         Rating: 8.8/10 
      </p>
       
      <p>
         Duration: 148 minutes 
      </p>
       
      <p>
         A thief who steals corporate secrets... 
      </p>
       
    </div>
     
  </div>
</DocumentFragment>
`;
